tests/Unit/Evolution/Systems/GeneralizedHarmonic/TestFunctions.py

```diff
@@ -184,3 +183,199 @@ def two_index_constraint(d_gauge_function, spacetime_normal_one_form,
     return constraint
 
 # End test functions for two-index constraint
+
+
+# Begin tests for Test_DampedHarmonicGaugeQuantities.cpp
+
+def weight_func(coords, r_max, r_width):
+    r2 = np.sum([coords[i]**2 for i in range(len(coords))])
+    return np.exp(- r2 * r_width / r_max / r_max)
+
+
+def spacetime_deriv_weight_func(coords, r_max, r_width):
+    W = weight_func(coords, r_max, r_width)
+    DW = np.zeros(len(coords) + 1)
+    DW[1:] = -2. * r_width * W / r_max / r_max * coords
+    return DW
+
+
+def roll_on_func(time, t_start, sigma_t):
+    if time < t_start: return 0.
+    return 1. - np.exp(- ((time - t_start)/sigma_t)**4 )
+
+
+def time_deriv_roll_on_func(time, t_start, sigma_t):
+    if time < t_start: return 0.
+    tnrm = (time - t_start)/sigma_t
+    return np.exp(- tnrm**4) * 4 * tnrm**3 / sigma_t
+
+def log_fac(lapse, sqrt_det_spatial_metric, exponent):
+    # if exponent == 0, the first term automatically vanishes
+    return exponent * np.log(sqrt_det_spatial_metric**2) - np.log(lapse)
+
+def spacetime_deriv_log_fac(lapse, shift, spacetime_unit_normal,
+        inverse_spatial_metric, sqrt_det_spatial_metric, dt_spatial_metric,
+        pi, phi, exponent):
+    SpatialDim = len(shift)
+    #
+    import os, imp
+    dname = os.path.abspath(os.path.join(os.path.dirname(\
+                os.path.realpath(__file__)),\
+                '../../../PointwiseFunctions/GeneralRelativity/'))
+    if not os.path.exists(dname):
+        dname = os.path.abspath(os.path.join(os.path.dirname(\
+                os.path.realpath(__file__)),\
+                '../../PointwiseFunctions/GeneralRelativity/'))
+    fname = os.path.join(dname, 'TestFunctions.py')
+    TF = imp.load_source('TestFunctions', fname)
```

Comment:
@osheamonn what do you think? Should we import python test functions across subdirectories? We think it could be better to just copy these test functions here.

Reply:
@geoffrey4444 has written code that does this but handled it in what I thought was a more elegant way. (I don't really love this use of relative paths). @geoffrey4444 could you please point to that code, I'm forgetting what exactly it was right now?

Reply:
@hrueter @nilsdeppe -
This is a hideous hacks that I had put in in order to access functions from other tests and refrain from code duplication. I was hoping the hackiness of this would generate some discussion on the broader question of how to do this in general. 

Going forward, if not already, I can imagine tests that would depend on basic functions written in (for e.g.) `PointwiseFunctions/GeneralRelativity`. Such tests *should* use Python versions of the corresponding functions that already exist in `tests/Unit/PointwiseFunctions/GeneralRelativity/TestFunctions.py` . 

Could we have a more systemic solution to this? Is @geoffrey4444 's solution generic and scalable? Could someone please point me to it?

Reply:
@nilsdeppe and myself discussed this recently and tentatively favour making a Python module out of `tests/`. We should discuss this in an issue and/or a call some time soon.

Reply:
It could be as simple as `touch __init__.py` in `tests` and subdirectories thereof..

Reply:
Please don't use git for this. I think nothing in /src or /tests should depend on git, except if it is testing the git infrastructure.

Can you import the necessary functions as it is done here?
[/tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/TestFunctions.py](https://github.com/sxs-collaboration/spectre/blob/develop/tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/TestFunctions.py)

Reply:
> Can you import the necessary functions as it is done here?
> [/tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/TestFunctions.py](https://github.com/sxs-collaboration/spectre/blob/develop/tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/TestFunctions.py)

Sure, I have changed to import `TestFunctions` module from `PointwiseFunctions/GeneralRelativity` directly. It works only if I have `$SPECTRE_HOME/tests/Unit/PointwiseFunctions`  in my PYTHONPATH.